Update About-Nodes.rst (#3917)
ros2/ros2_documentation: source/Concepts/Basic/About-Nodes.rst

Nodes can :doc:`publish <About-Topics>` to named topics to deliver data to other nodes, or :doc:`subscribe <About-Topics>` to named topics to get data from other nodes.

> They can also act as a :doc:`service client <About-Services>` to have another node perform a computation on their behalf, or as a :doc:`service server <About-Services>` to provide functionality to other nodes.

Reviewer: ```suggestion
They can also use a :doc:`service client <About-Services>` to have another node perform a computation on their behalf, or implement a :doc:`service server <About-Services>` to provide that functionality to other nodes.
```

Author: Thanks for your feedback. I believe that the term 'use/implement', can make it seem like they are using another object to do something but in fact, it is the intrinsic property of a node to behave/act a certain way. I prefer using the former sentence structure; was able to understand the concept of a node's diverse nature pretty well. Thanks!

Reviewer: > Thanks for your feedback. I believe that the term 'use/implement', can make it seem like they are using another object to do something but in fact, it is the intrinsic property of a node to behave/act a certain way. I prefer using the former sentence structure; was able to understand the concept of a node's diverse nature pretty well. Thanks!

The thing is, I don't think that is the case.  A node can contain publishers, subscribers, service servers, service clients, action servers, and action clients, all at the same time.  That is, a node is really a container for these other things, and in that sense nodes can be a bunch of different things at once, so I think it is appropriate to point that out here.

That said, I also don't love the terminology of "use/implement".  I'll try to think of another way to present this, but any suggestions are welcome.

> For long-running computations, a node can act as an :doc:`action server <About-Actions>` to perform it, or as an :doc:`action client <About-Actions>` to have another node perform it.

Reviewer: ```suggestion
For long-running computations, a node can use an :doc:`action client <About-Actions>` to have another node perform the action, or as an :doc:`action server <About-Actions>` to actually implement the action.
```

Author: Okay, that makes sense. I'm from a DevOps background and new to ROS, and the way you explained that makes it a lot more clear. I still believe, out of the given options, 'act' is the closest in meaning, and considering the above paragraph of the docs and hyperlinks, I'd keep it consistent.  We can add another paragraph saying - 

Oftentimes, nodes are a complex combination of a bunch of things.  A node can contain a combination of publishers, subscribers, service servers, service clients, action servers, and action clients, all at the same time and based on the required functionality, they can use the required feature. 
What do you think?

Reviewer: OK, let's add that paragraph.  I think I'll still want to update the language a bit, but let's add that paragraph first and then I'll do another review.  Thanks!

Nodes can :doc:`publish <About-Topics>` to named topics to deliver data to other nodes, or :doc:`subscribe <About-Topics>` to named topics to get data from other nodes.
They can also act as a :doc:`service client <About-Services>` to have another node perform a computation on their behalf, or as a :doc:`service server <About-Services>` to provide functionality to other nodes.

> For long-running computations, a node can act as an :doc:`action server <About-Actions>` to perform it, or as an :doc:`action client <About-Actions>` to have another node perform it.

Reviewer: ```suggestion
For long-running computations, a node can act as an :doc:`action client <About-Actions>` to have another node perform it on their behalf, or as an :doc:`action server <About-Actions>` to provide functionality to other nodes.
```

Nodes can provide configurable :doc:`parameters <About-Parameters>` to change behavior during run-time.

> Oftentimes, nodes are a complex combination of a bunch of things. A node can contain a combination of publishers, subscribers, service servers, service clients, action servers, and action clients, all at the same time and based on the required functionality, they can use the required feature.

Reviewer: ```suggestion
Nodes are often a complex combination of publishers, subscribers, service servers, service clients, action servers, and action clients, all at the same time.
```
